Code does not obey Pylint rules

When using the sample code snippet pylint (in default configuration) complains about some issues:

>************* Module server
server.py:170:0: C0301: Line too long (127/100) (line-too-long)
server.py:24:0: C0115: Missing class docstring (missing-class-docstring)
server.py:25:4: W0231: __init__ method from base class 'Exception' is not called (super-init-not-called)
server.py:31:0: C0116: Missing function or method docstring (missing-function-docstring)
server.py:48:4: R1720: Unnecessary "elif" after "raise" (no-else-raise)
server.py:81:0: C0103: Argument name "f" doesn't conform to snake_case naming style (invalid-name)
server.py:92:12: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
server.py:121:16: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
server.py:124:16: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
server.py:129:16: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
> ------------------------------------------------------------------
>Your code has been rated at 8.84/10

--- a/00-Starter-Seed/server.py
+++ b/00-Starter-Seed/server.py
@@ -22,6 +22,9 @@ APP = Flask(__name__)
 
 # Format error response and append status code.
 class AuthError(Exception):
+    """
+    An AuthError is raised whenever the authentication failed.
+    """
     def __init__(self, error, status_code):
         super().__init__()
         self.error = error

--- a/00-Starter-Seed/server.py
+++ b/00-Starter-Seed/server.py
@@ -23,6 +23,7 @@ APP = Flask(__name__)
 # Format error response and append status code.
 class AuthError(Exception):
     def __init__(self, error, status_code):
+        super().__init__()
         self.error = error
         self.status_code = status_code
 

--- a/00-Starter-Seed/server.py
+++ b/00-Starter-Seed/server.py
@@ -33,6 +33,11 @@ class AuthError(Exception):
 
 @APP.errorhandler(AuthError)
 def handle_auth_error(ex):
+    """
+    serializes the given AuthError as json and sets the response status code accordingly.
+    :param ex: an auth error
+    :return: json serialized ex response
+    """
     response = jsonify(ex.error)
     response.status_code = ex.status_code
     return response

--- a/00-Starter-Seed/server.py
+++ b/00-Starter-Seed/server.py
@@ -51,10 +51,10 @@ def get_token_auth_header():
                         "description":
                             "Authorization header must start with"
                             " Bearer"}, 401)
-    elif len(parts) == 1:
+    if len(parts) == 1:
         raise AuthError({"code": "invalid_header",
                         "description": "Token not found"}, 401)
-    elif len(parts) > 2:
+    if len(parts) > 2:
         raise AuthError({"code": "invalid_header",
                         "description":
                             "Authorization header must be"

--- a/00-Starter-Seed/server.py
+++ b/00-Starter-Seed/server.py
@@ -79,10 +79,10 @@ def requires_scope(required_scope):
     return False
 
 
-def requires_auth(f):
+def requires_auth(func):
     """Determines if the access token is valid
     """
-    @wraps(f)
+    @wraps(func)
     def decorated(*args, **kwargs):
         token = get_token_auth_header()
         jsonurl = urlopen("https://"+AUTH0_DOMAIN+"/.well-known/jwks.json")
@@ -133,7 +133,7 @@ def requires_auth(f):
                                     " token."}, 401)
 
             _request_ctx_stack.top.current_user = payload
-            return f(*args, **kwargs)
+            return func(*args, **kwargs)
         raise AuthError({"code": "invalid_header",
                         "description": "Unable to find appropriate key"}, 401)
     return decorated

--- a/00-Starter-Seed/server.py
+++ b/00-Starter-Seed/server.py
@@ -89,11 +89,11 @@ def requires_auth(func):
         jwks = json.loads(jsonurl.read())
         try:
             unverified_header = jwt.get_unverified_header(token)
-        except jwt.JWTError:
+        except jwt.JWTError as jwt_error:
             raise AuthError({"code": "invalid_header",
                             "description":
                                 "Invalid header. "
-                                "Use an RS256 signed JWT Access Token"}, 401)
+                                "Use an RS256 signed JWT Access Token"}, 401) from jwt_error
         if unverified_header["alg"] == "HS256":
             raise AuthError({"code": "invalid_header",
                             "description":
@@ -118,19 +118,19 @@ def requires_auth(func):
                     audience=API_IDENTIFIER,
                     issuer="https://"+AUTH0_DOMAIN+"/"
                 )
-            except jwt.ExpiredSignatureError:
+            except jwt.ExpiredSignatureError as expired_sign_error:
                 raise AuthError({"code": "token_expired",
-                                "description": "token is expired"}, 401)
-            except jwt.JWTClaimsError:
+                                "description": "token is expired"}, 401) from expired_sign_error
+            except jwt.JWTClaimsError as jwt_claims_error:
                 raise AuthError({"code": "invalid_claims",
                                 "description":
                                     "incorrect claims,"
-                                    " please check the audience and issuer"}, 401)
-            except Exception:
+                                    " please check the audience and issuer"}, 401) from jwt_claims_error
+            except Exception as exc:
                 raise AuthError({"code": "invalid_header",
                                 "description":
                                     "Unable to parse authentication"
-                                    " token."}, 401)
+                                    " token."}, 401) from exc
 
             _request_ctx_stack.top.current_user = payload
             return func(*args, **kwargs)

